CMakeLists.txt

```diff
@@ -144,16 +152,31 @@ if (LIBDDWAF_BUILD_STATIC)
             COMMAND ${CMAKE_COMMAND} -E copy lib_comb/combined${CMAKE_STATIC_LIBRARY_SUFFIX} $<TARGET_FILE:libddwaf_static>
             WORKING_DIRECTORY ${CMAKE_BINARY_DIR})
     else()
-        add_custom_command(TARGET libddwaf_static POST_BUILD
-            COMMAND ${CMAKE_COMMAND} -E make_directory ar_comb
-            COMMAND ${CMAKE_COMMAND} -E chdir ar_comb ${CMAKE_AR} -x $<TARGET_FILE:libddwaf_static>
-            COMMAND ${CMAKE_COMMAND} -E chdir ar_comb ${CMAKE_AR} -x $<TARGET_FILE:lib_re2>
-            COMMAND ${CMAKE_COMMAND} -E chdir ar_comb ${CMAKE_AR} -x $<TARGET_FILE:lib_ac>
-            COMMAND ${CMAKE_COMMAND} -E chdir ar_comb ${CMAKE_AR} -x $<TARGET_FILE:lib_injection>
-            COMMAND cd ar_comb && ${CMAKE_AR} -qcs combined${CMAKE_STATIC_LIBRARY_SUFFIX} *.o*
-            COMMAND ${CMAKE_COMMAND} -E copy ar_comb/combined${CMAKE_STATIC_LIBRARY_SUFFIX} $<TARGET_FILE:libddwaf_static>
-            COMMAND rm -rf ar_comb
-            WORKING_DIRECTORY ${CMAKE_BINARY_DIR})
+        if(CMAKE_OSX_ARCHITECTURES MATCHES "arm64")
+          add_custom_command(TARGET libddwaf_static POST_BUILD
+              COMMAND ${CMAKE_COMMAND} -E make_directory ar_comb
+              COMMAND ${CMAKE_COMMAND} -E chdir ar_comb ${CMAKE_AR} -x $<TARGET_FILE:libddwaf_static>
+              COMMAND ${CMAKE_COMMAND} -E chdir ar_comb ${CMAKE_AR} -x $<TARGET_FILE:lib_re2>
+              COMMAND ${CMAKE_COMMAND} -E chdir ar_comb ${CMAKE_AR} -x $<TARGET_FILE:lib_ac>
+              COMMAND ${CMAKE_COMMAND} -E chdir ar_comb ${CMAKE_AR} -x $<TARGET_FILE:lib_injection>
+              COMMAND cd ar_comb && libtool -static -o combined.a -arch_only ${CMAKE_OSX_ARCHITECTURES} *.o*
```

Comment:
So this case uses `libtool` with an additional flag. Maybe `libtool` could be used on all platforms instead of `ar`, which would help in getting rid of the big `if()`?

Reply:
It's not pretty, but I think it's preferable to require that linux systems have `libtool` installed. Another option would be to try to get rid of the `else` by using `-Wl,--whole-archive`, which I don't think is supported in mac os.